graphql_compiler/tests/test_input_data.py

```diff
@@ -2431,3 +2431,37 @@ def complex_nested_optionals():
         expected_output_metadata=expected_output_metadata,
         expected_input_metadata=expected_input_metadata,
         type_equivalence_hints=None)
+
+
+def no_op_coercion():
```

Comment:
nit: mind moving this test input (and its associated test case) to the place in the inputs/tests files where type coercions are being tested? Grouping similar tests together helps with refactoring and looking for similar issues.